src/onnx/parse_softmaxcrossentropyloss.cpp

```diff
@@ -0,0 +1,308 @@
+            std::vector<float> zero_val_vect(weights_shape.elements(), 0);
+            auto zero_val = info.add_literal(migraphx::literal(weights_shape, zero_val_vect));
+            weights       = info.add_instruction(
+                migraphx::make_op("scatter_none", {{"axis", 0}}), weights, ignore_index, zero_val);
+        }
+
+        if(is_k_dim)
+        {
+            auto lens = label_shape.lens();
+            size_t d =
+                std::accumulate(lens.rbegin(), lens.rend() - 1, 1, std::multiplies<size_t>());
+            scores = info.add_instruction(
+                migraphx::make_op("reshape", {{"dims", {batch_size, class_size, d}}}), scores);
+        }
+
+        // Offload calculation of log(Softmax(scores)) for the input before we perform cross entropy
+        // loss calculation
+        auto softmax_scores = info.add_instruction(migraphx::make_op("softmax"), scores);
+        auto log_sm_scores  = info.add_instruction(migraphx::make_op("log"), softmax_scores);
+        auto neg_lsm_scores = info.add_instruction(migraphx::make_op("neg"), log_sm_scores);
+
+        loss_tensor = info.add_instruction(
+            migraphx::make_op("scatter_none", {{"axis", 0}}), loss_tensor, labels, neg_lsm_scores);
+
+        if(is_k_dim)
+        {
+            loss_tensor = info.add_instruction(
+                migraphx::make_op("reshape", {{"dims", label_shape.lens()}}), loss_tensor);
+        }
+
+        instruction_ref weight_tensor;
+        if(has_ignore_index or has_weights)
+        {
+            weight_tensor =
+                info.add_instruction(migraphx::make_op("gather", {{"axis", 0}}), weights, labels);
+
+            // Mul op also handles ignored index as the corresponding weight will be zero if it
+            // matches ignore index
+            loss_tensor =
+                info.add_instruction(migraphx::make_op("mul"), loss_tensor, weight_tensor);
+        }
+
+        // Add reduction step after we're generated crossentropyloss tensor and rearragned weight
+        // scaling tensor
+        if(reduction == "mean" and not has_weights)
+        {
+            loss_tensor = info.add_instruction(migraphx::make_op("reduce_mean", {{"axes", {0}}}),
+                                               loss_tensor);
+        }
+        else if(reduction == "sum" or has_weights)
+        {
+            if(reduction != "none")
+            {
+                loss_tensor = info.add_instruction(migraphx::make_op("reduce_sum", {{"axes", {0}}}),
+                                                   loss_tensor);
+            }
+
+            if(reduction == "mean")
+            {
+                auto reduced_weights = info.add_instruction(
+                    migraphx::make_op("reduce_sum", {{"axes", {0}}}), weight_tensor);
+                loss_tensor =
+                    info.add_instruction(migraphx::make_op("div"), loss_tensor, reduced_weights);
+            }
+        }
```

Comment:
Better to first handle 
`reduction == mean and has_weights` case. 
then you can do do `reduction == mean` case
`reduction == sum` case 
and as the default case just return "None" case.

Similar to onnx ref implementation

Reply:
Ah I can reorder this ok.

Reply:
Done